packages/indexer-agent/src/agent.ts

```diff
@@ -688,8 +689,11 @@ class Agent {
     // Index all new deployments worth indexing
     await queue.addAll(
       deploy.map(deployment => async () => {
-        const name = `indexer-agent/${deployment.ipfsHash.slice(-10)}`
-
+        const subgraphDeployment =
+          await this.networkMonitor.requireSubgraphDeployment(
```

Comment:
That method was introduced because the `formatDeploymentName` function requires a `SubgraphDeployment` so it has all the needed data to produce the subgraph deployment name. 

It's worth noting that we are also introducing a call to the Network Subgraph at this point.

Reply:
The `deployment` variable is a `SubgraphDeploymentID`, which doesn't have the required info _(subgraph and creator names)_ to create the deployment name.

The `NetworkMonitor.requireSubgraphDeployment` method uses that ID to query for that extra data and returns the necessary `SubgraphDeployment` value.